src/boot/bootctl.c

```diff
@@ -1647,6 +1653,18 @@ static int parse_argv(int argc, char *argv[]) {
                         arg_arch_all = true;
                         break;
 
+                case ARG_EFI_LABEL:
+                        if (isempty(optarg) || !string_is_safe(optarg))
+                                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                                       "Invalid --efi-label=: %s", strna(optarg));
+                        if (strlen(optarg) > 1024)
+                                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                                       "--efi-label= too long: %zu > 1024", strlen(optarg));
+                        r = free_and_strdup(&arg_efi_label, optarg);
+                        if (r < 0)
+                                return log_oom();
```

Comment:
Not a matter, but you can use `free_and_strdup_warn()`:
```
return free_and_strdup_warn(&arg_efi_label, optarg);
```

Reply:
I can't return here because if there is no error during processing all other parameters would be skipped.

This would work, but not sure if worth?
```
                        r = free_and_strdup_warn(&arg_efi_label, optarg);
                        if (r < 0)
                                return r;
```

Reply:
Indeed.